docs/DeveloperGuide.md

```diff
@@ -6,7 +6,154 @@
+- `void addDish(Dish dish)` -- add a new dish.
+- `void deleteDish(int menuIndex)` -- delete a dish using its menu index.
+- `void setPrice(int menuIndex, double newPrice)` -- change the price of a dish using its menu index.
+- `void setName(int menuIndex, String newName)` -- change the name of a dish using its menu index.
+
+These functions are "delegated" to `ArrayList<Dish>`, which is associated with `DishManager`.
+
+For example, `printDishes()`, iterates through `dishes` and calls `get(int)` method (`setPrice(int, double)`, `setName(int, String)` also call `get(int)`):
+
+![](diagrams/sqeuence/DishManager_printDishes.png)
+
+`deleteDish(int)` calls `remove(int)`:
+
+![](diagrams/sqeuence/DishManager_deleteDish.png)
+
+`addDish(Dish)` calls `add(Dish)`:
+
+![](diagrams/sqeuence/DishManager_addDish.png)
+
+
+
+### Order Management
+
+The management of menu is functioned by `OrderManager`, which implements following methods:
+- `void addDishToOrder(Dish dish, int orderIdx)` -- add dishes in to an order.
+- `void deleteOrder(int orderIndex)` -- delete an order using its order index.
+- `Order getOrder(int orderIndex)` -- get an order using its order index.
+- `double getOrderPrice(int orderIndex)` -- get the total price of an order using its order index.
+- `double getAllOrderValue()` -- get the total price of all orders.
+
+These functions are "delegated" to `ArrayList<Order>`, which is associated with `OrderManager`. 
+
+Methods call `get(int)` to get the element.
+
+For example, `addDishToOrder(Dish dish, int orderIdx)`, iterates through `orders` and calls `addDish(Dish dish)` method:
+
+![](diagrams/sqeuence/OrderManager_addDishToOrder.png)
+
+`deleteOrder(int)` calls `getOrderCount()`:
+![](diagrams/sqeuence/OrderManager_deleteOrder.png)
+
+Both methods `getOrderPrice(int)` and `getAllOrderValue()` call `getTotalPrice()`:
+![](diagrams/sqeuence/OrderManager_displayAllOrderPrice.png)
+
+### Staff Management
+
+The management of staff is handled by `StaffManager`, which implements following methods:
+
+- `void printStaff()` -- print all staffs.
+- `void addStaff(int staffId, String staffName, String position, double salary)` -- add a new staff with the arguments as its attributes.
+- `Staff findByStaffId(int staffId, boolean printMsg)` -- find a staff using staff id, and print message if boolean variable printMsg is true.
+- `void deleteByStaffId(int staffId)` -- delete a staff using staff id.
+- `int getNumOfStaffs(` -- change the name of a dish using its menu index.
+
+`StaffManager` manages `ArrayList<Staff>` and the above functions help to update the collection of `Staff`.
+
+For example `addStaff(int staffId, String staffName, String position, double salary)` calls `add(Staff)` to add a `Staff` from `ArrayList<Staff>`:
+
+![](diagrams/sqeuence/StaffManager_addStaff.png)
+
+And `deleteStaff(int)` calls `remove(int)` to remove a `Staff` from `ArrayList<Staff>`:
+
+![](diagrams/sqeuence/StaffManager_deleteStaff.png)
```

Comment:
<img width="637" alt="Screen Shot 2022-04-01 at 10 30 09 AM" src="https://user-images.githubusercontent.com/73009417/161183429-b42523d0-11f4-4bb0-a17c-870d16cb6a4f.png">
Is it better to add a return arrow below the 'opt' box? Since it shows the normal flow when staff has been found.